Require `ContentHandlerError` to implement Send and Sync

I tried using `rewrite_str` with `failure`, but it doesn't implement Send and Sync:

```
error[E0277]: `(dyn std::error::Error + 'static)` cannot be sent between threads safely
   --> src/utils/html.rs:47:5
    |
47  |     lol_html::rewrite_str(html, settings).map_err(Into::into)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn std::error::Error + 'static)` cannot be sent between threads safely
    | 
   ::: /home/joshua/.local/lib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/convert/mod.rs:288:1
    |
288 | pub trait Into<T>: Sized {
    | ------------------------ required by this bound in `std::convert::Into::into`
    |
    = help: the trait `std::marker::Send` is not implemented for `(dyn std::error::Error + 'static)`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::ptr::Unique<(dyn std::error::Error + 'static)>`
    = note: required because it appears within the type `std::boxed::Box<(dyn std::error::Error + 'static)>`
    = note: required because it appears within the type `lol_html::rewriter::RewritingError`
    = note: required because of the requirements on the impl of `failure::Fail` for `lol_html::rewriter::RewritingError`
    = note: required because of the requirements on the impl of `std::convert::From<lol_html::rewriter::RewritingError>` for `failure::error::Error`
    = note: required because of the requirements on the impl of `std::convert::Into<failure::error::Error>` for `lol_html::rewriter::RewritingError`

error[E0277]: `(dyn std::error::Error + 'static)` cannot be shared between threads safely
   --> src/utils/html.rs:47:5
    |
47  |     lol_html::rewrite_str(html, settings).map_err(Into::into)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn std::error::Error + 'static)` cannot be shared between threads safely
    | 
   ::: /home/joshua/.local/lib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/convert/mod.rs:288:1
    |
288 | pub trait Into<T>: Sized {
    | ------------------------ required by this bound in `std::convert::Into::into`
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn std::error::Error + 'static)`
    = note: required because of the requirements on the impl of `std::marker::Sync` for `std::ptr::Unique<(dyn std::error::Error + 'static)>`
    = note: required because it appears within the type `std::boxed::Box<(dyn std::error::Error + 'static)>`
    = note: required because it appears within the type `lol_html::rewriter::RewritingError`
    = note: required because of the requirements on the impl of `failure::Fail` for `lol_html::rewriter::RewritingError`
    = note: required because of the requirements on the impl of `std::convert::From<lol_html::rewriter::RewritingError>` for `failure::error::Error`
    = note: required because of the requirements on the impl of `std::convert::Into<failure::error::Error>` for `lol_html::rewriter::RewritingError`

error: aborting due to 2 previous errors
```

failure has [a workaround](https://docs.rs/failure/0.1.8/failure/struct.SyncFailure.html) for missing Sync (i.e. a mutex), but not for missing `Send`. It would be great if `ContentHandlerError` implemented both. It looks like this is a user-provided error, so this would be a breaking change, but I think the increase in flexibility is worth it.

(Yes, I'm aware `failure` is deprecated, but docs.rs won't be switching any time soon and this is useful even without failure)

--- a/src/rewriter/mod.rs
+++ b/src/rewriter/mod.rs
@@ -72,7 +72,7 @@ pub enum RewritingError {
 
     /// An error that was propagated from one of the content handlers.
     #[error("{0}")]
-    ContentHandlerError(Box<dyn StdError>),
+    ContentHandlerError(Box<dyn StdError + Send + Sync>),
 }
 
 /// A streaming HTML rewriter.

--- a/src/rewriter/settings.rs
+++ b/src/rewriter/settings.rs
@@ -2,7 +2,7 @@ use crate::rewritable_units::{Comment, Doctype, DocumentEnd, Element, EndTag, Te
 use crate::selectors_vm::Selector;
 use std::error::Error;
 
-pub(super) type HandlerResult = Result<(), Box<dyn Error>>;
+pub(super) type HandlerResult = Result<(), Box<dyn Error + Send + Sync>>;
 pub type DoctypeHandler<'h> = Box<dyn FnMut(&mut Doctype) -> HandlerResult + 'h>;
 pub type CommentHandler<'h> = Box<dyn FnMut(&mut Comment) -> HandlerResult + 'h>;
 pub type TextHandler<'h> = Box<dyn FnMut(&mut TextChunk) -> HandlerResult + 'h>;

